pkg/cluster/clusterserviceprincipal.go

```diff
@@ -173,50 +177,50 @@ func (m *manager) updateAROSecret(ctx context.Context) error {
 }
 
 func (m *manager) updateOpenShiftSecret(ctx context.Context) error {
-	var changed bool
-	spp := m.doc.OpenShiftCluster.Properties.ServicePrincipalProfile
-	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
We'll know it fails because we're returing any error after the `Apply` on line 195. It's less "passing the buck" to the server and more using a feature that's explicity intended to simplify controller logic and make them more declarative ((ref)[https://kubernetes.io/docs/reference/using-api/server-side-apply/#comparison-with-client-side-apply]). Part of the simplification of logic is that we no longer have to worry about retrying on conflict, because the server is just going to force our changes on a conflict. It's the same end result.